### ISSUE TITLE
Not working on ST2 build 2221 OS X 10.6.7

### ISSUE THREAD

**Issue by fcccbf8f8f1** [wbond/package_control#1162]
Can't access package control via command palette. I've turned debug on, but nothing is being logged because package control is listed in in_process_packages.

some of the error logs indicate that the OS must be 7 or HIGHER, yet I only have 10.6.8 and everything was working fine until last night. Please restore support for 10.6.8.

Here are the relevant error logs)  from start up.

console error logs Package COntrol

```
Reloading plugin /Users/fcccbf8f8f1/Library/Application Support/Sublime Text 2/Packages/Open Folder/OpenFolder.py
Reloading plugin /Users/fcccbf8f8f1/Library/Application Support/Sublime Text 2/Packages/Package Control/1_reloader.py
Reloading plugin /Users/fcccbf8f8f1/Library/Application Support/Sublime Text 2/Packages/Package Control/2_bootstrap.py
Traceback (most recent call last):
  File "./sublime_plugin.py", line 62, in reload_plugin
  File "./2_bootstrap.py", line 22, in <module>
    from package_control.bootstrap import bootstrap_dependency, mark_bootstrapped
  File "./package_control/bootstrap.py", line 20, in <module>
    from .download_manager import downloader
  File "./package_control/download_manager.py", line 24, in <module>
    from .downloaders import DOWNLOADERS
  File "./package_control/downloaders/__init__.py", line 3, in <module>
    from .urllib_downloader import UrlLibDownloader
  File "./package_control/downloaders/urllib_downloader.py", line 46, in <module>
    from ..ca_certs import get_ca_bundle_path
  File "./package_control/ca_certs.py", line 7, in <module>
    from .deps.oscrypto import trust_list
  File "./package_control/deps/oscrypto/trust_list.py", line 19, in <module>
    from ._osx.trust_list import extract_from_system, system_path
  File "./package_control/deps/oscrypto/_osx/trust_list.py", line 9, in <module>
    from ._security import Security, SecurityConst, handle_sec_error
  File "./package_control/deps/oscrypto/_osx/_security.py", line 10, in <module>
    from ._security_ctypes import Security, version_info as osx_version_info
  File "./package_control/deps/oscrypto/_osx/_security_ctypes.py", line 24, in <module>
    raise OSError('Only OS X 10.7 and newer are supported, not %s.%s' % (version_info[0], version_info[1]))
OSError: Only OS X 10.7 and newer are supported, not 10.6
Reloading plugin /Users/fcccbf8f8f1/Library/Application Support/Sublime Text 2/Packages/Package Control/Package Control.py
Traceback (most recent call last):
  File "./sublime_plugin.py", line 62, in reload_plugin
  File "./Package Control.py", line 95, in <module>
    from package_control.commands import *  # noqa
  File "./package_control/commands/__init__.py", line 3, in <module>
    from .advanced_install_package_command import AdvancedInstallPackageCommand
  File "./package_control/commands/advanced_install_package_command.py", line 10, in <module>
    from ..package_manager import PackageManager
  File "./package_control/package_manager.py", line 34, in <module>
    from .downloaders.background_downloader import BackgroundDownloader
  File "./package_control/downloaders/__init__.py", line 3, in <module>
    from .urllib_downloader import UrlLibDownloader
  File "./package_control/downloaders/urllib_downloader.py", line 46, in <module>
    from ..ca_certs import get_ca_bundle_path
  File "./package_control/ca_certs.py", line 7, in <module>
    from .deps.oscrypto import trust_list
  File "./package_control/deps/oscrypto/trust_list.py", line 19, in <module>
    from ._osx.trust_list import extract_from_system, system_path
  File "./package_control/deps/oscrypto/_osx/trust_list.py", line 9, in <module>
    from ._security import Security, SecurityConst, handle_sec_error
  File "./package_control/deps/oscrypto/_osx/_security.py", line 10, in <module>
    from ._security_ctypes import Security, version_info as osx_version_info
  File "./package_control/deps/oscrypto/_osx/_security_ctypes.py", line 24, in <module>
    raise OSError('Only OS X 10.7 and newer are supported, not %s.%s' % (version_info[0], version_info[1]))
OSError: Only OS X 10.7 and newer are supported, not 10.6
```

**Comment by wbond** [comment]
The library used for exported TLS certificates from the trust store uses the OS X Security framework. Unfortunately that framework in version 10.6 is missing a number of functions that the oscrypto library uses.

It may be possible to find a way to use alternative functions in 10.6, but as of right now I am not willing to spend my free time on that, so there would need to be a fundraiser to raise enough funds for me to spend some freelance time on that rather than some other paying work I have.

It may make more sense for you to use git and clone version 3.1.2 and stay on that until you decide to upgrade your OS.

**Comment by astiob** [comment]
At the very least, I think it would be nice if Package Control didn’t automatically upgrade itself (or let the user upgrade it) to the latest version on 10.6.

**Comment by wbond** [comment]
@astiob There is currently no way to filter by OS version in Package Control, so the label major enhancement still applies.

Just an FYI, OS X 10.6 is pretty insecure at this point.

**Comment by rteves18** [comment]
Hello @wbond long story short, I'm also running on OS X 10.6.8 and due to hardware issues, unfortunately it is out of the question to update the OS on my machine at the moment. I'm fairly new to github so I apologize for the silly question, but when you say clone version 3.1.2, what exact commit id/repo should I `git clone` under what exact directory? Thank you for your time.

**Comment by astiob** [comment]
For what it’s worth, I resolved this (I hope) by removing Package Control from `~/Library/Application Support/Sublime Text 2/Packages` and replacing it with the last good copy from `~/Library/Application Support/Sublime Text 2/Backup` (which in my case was dated 20160921163710), and adding `"auto_upgrade_ignore": ["Package Control"]` to my user settings of Package Control.

Is this worse than checking out 3.1.2 from Git or is it the same?

To use Git, I think you just need to `git clone -b 3.1.2 https://github.com/wbond/package_control.git "Package Control"` within the `Packages` directory.

**Comment by wbond** [comment]
@rteves18 That is fine that you have old hardware. I just can't spend extra time to try and accommodate that, or sacrifice the security of users who do run modern OSes that do receive security patches.

Yes, on GitHub there is a "Releases" tab that has a sub-tab for tags that points to https://github.com/wbond/package_control/tags. Here you can download a zip of any old version.

Just be aware that over time those old versions will stop working as the channel server moves forward. At some point you will be left with a version of Package Control that does not work.

**Comment by NealeTools** [comment]
I am also very new to github (and to coding). I downloaded the 3.1.2 file, renamed it "Package Control" and copied int the existing packages folder (overwriting the newer version). I added the following text within the curly brackets of the user settings within sublime: "auto_upgrade_ignore": ["Package Control"].

It seemed to work...Package Control commands now appeared in the Command Palette, although choosing Install Package did nothing...

I restarted ST2 and now it looks Package Control has been updated to the newer version already and no longer works in the command palette. Any ideas?

The only package I need to add is Sublime REPL. Is there any way other than using Package Control that this can be achieved?
